src/client/app/components/ThreeDPillComponent.tsx

```diff
@@ -43,18 +43,48 @@ export default function ThreeDPillComponent() {
 		return { meterOrGroupID: groupID, isDisabled: isDisabled, meterOrGroup: MeterOrGroup.groups } as MeterOrGroupPill;
 	});
 
-	// when there is only one choice, it must be selected as a default (there is no other option)
+	// Merge meters and groups into one array
+	const combinedPillData = [...meterPillData, ...groupPillData];
+	// Track length of combinedPillData to determine if a new pill was added
+	const prevLengthRef = React.useRef(combinedPillData.length);
+	// Anonymous selector to get the last added is a meter or group
+	const lastAddedType = useAppSelector(state => state.graph.current.lastAddedMeterOrGroup);
 	useEffect(() => {
-		const combinedPillData = [...meterPillData, ...groupPillData];
-
+		// If only one item is selected, auto-select it
 		if (combinedPillData.length === 1) {
 			const singlePill = combinedPillData[0];
 			dispatch(updateThreeDMeterOrGroupInfo({
 				meterOrGroupID: singlePill.meterOrGroupID,
 				meterOrGroup: singlePill.meterOrGroup
 			}));
 		}
-	}, [meterPillData, groupPillData]);
+		// If the current 3D selection is no longer present, clear it
+		const stillExists = combinedPillData.some(
+			pill => pill.meterOrGroupID === threeDState.meterOrGroupID && pill.meterOrGroup === threeDState.meterOrGroup
+		);
+		if (!stillExists && (threeDState.meterOrGroupID !== undefined)) {
+			dispatch(updateThreeDMeterOrGroupInfo({ meterOrGroupID: undefined, meterOrGroup: undefined }));
+		}
+		// Auto-select new item on 3D chart if a new pill was added
+		if (
+			combinedPillData.length > prevLengthRef.current &&
+			lastAddedType
+		) {
+			let lastAdded;
+			if (lastAddedType === MeterOrGroup.meters && meterPillData.length > 0) {
```

Comment:
Could you add a comment to explain the logic (why does meter have to check length, etc.).? Thanks.